### PR TITLE
[Bug] [misc] Use sys.getfilesystemencoding() to prevent possible locale errors

### PR DESCRIPTION
**Pull request by archibate**
Related issue = https://github.com/taichi-dev/taichi_three/issues/13 https://github.com/taichi-dev/taichi/issues/428

See https://www.python.org/dev/peps/pep-0529/#update-sys-getfilesystemencoding.

[[Click here for the format server]](http://kun.csail.mit.edu:31415/)

----
@k-ye Would you help me confirm this works on OS X for all `locale` combinations?

### CODE REVIEW

File: python/taichi/core/util.py [hunk 1]
```diff
@@ -44,17 +44,18 @@ def import_ti_core(tmp_dir=None):
     if get_os_name() != 'win':
         sys.setdlopenflags(old_flags)
     lib_dir = os.path.join(package_root(), 'lib')
+    tmp_dir = '彭于斌'
```
Comment (archibate) [inline python/taichi/core/util.py:47]:
```suggestion
```

Comment (k-ye) [reply python/taichi/core/util.py:47]: remove?

**Review by k-ye — APPROVED** [review]
LGTM

**Comment by archibate** [comment]
> Do you know how I can repro this?

I'm not sure, what's your MacOS and Python version? And the one in taichi_three#13?

**Comment by archibate** [comment]
Locales are always international packages' headache yet no way to test against them at the time being. Let's try to merge this in given that CI passed and we used enough try-except's here and hopefully it can solve the hidden locale problems :)

> but I don't have this problem.

Actually are you testing in `master` without this PR?

**Pull request merged**